dwave/optimization/_model.pyx

```diff
@@ -1567,6 +1575,46 @@ cdef class ArraySymbol(Symbol):
             return out
         raise ValueError("only integer exponents of 1 or greater are supported")
 
+    def __radd__(self, lhs):
+        try:
+            lhs = _as_array_symbol(self.model, lhs)
+        except TypeError:
+            return NotImplemented
+
+        return lhs + self
```

Comment:
For the commutative operators, can we avoid the extra `_as_array_symbol()` call here and just `return self + lhs`, letting `__add__` take care of it?